### DISCUSSION TITLE
Message passing propagate method

### DISCUSSION THREAD

**Discussion by Angelo92Git** [pyg-team/pytorch_geometric#9058] [Q&A]
I'm a beginner getting familiar with pytorch geometric and I'm getting stuck with something basic when I try to create a custom message passing layer. The documentation doesn't appear to help. (torch_geometric version 2.5.1)

I was just trying to go follow the tutorial: Creating Message Passing Networks:

https://pytorch-geometric.readthedocs.io/en/latest/tutorial/create_gnn.html

But the propagate method (called in the forward method defined under the GCNConv class) keeps triggering an error:
```
# Step 4-5: Start propagating messages
out = self.propagate(edge_index, x=x, norm=norm)
```
Resulting in:
```
TypeError: propagate() got an unexpected keyword argument 'x'
```

The documentation for the propagate method suggests that it should be able to accept keyword arguments, so I am uncertain why this is erroring. I am using the code provided in the tutorial page (linked above) on a very simple graph.
https://pytorch-geometric.readthedocs.io/en/latest/generated/torch_geometric.nn.conv.MessagePassing.html?highlight=propagate

Please let me know if you have any insights as to what the issue is.

**Comment by aliwister** [comment]
I ran into the same problem using the latest release of PyG. I downgraded to 2.4, and it worked fine:

pip install torch-geometric==2.4.0

**Comment by rusty1s** [comment]
Lets move discussion to https://github.com/pyg-team/pytorch_geometric/issues/9059
